src/frame/sdk/GroupApi.ts

```diff
@@ -145,6 +152,33 @@ export function create(
         .map((createdGroup) => formatDetailedGroupResponse(createdGroup) as GroupDetailResponse);
 }
 
+/**
+ * Rotate a groups current private key
+ */
+export function rotateGroupPrivateKey(groupID: string) {
+    const {privateKey} = ApiState.deviceKeys();
+    return GroupApiEndpoints.callGroupGetApi(groupID).flatMap((group) => {
+        if (!isGroupAdminResponse(group)) {
+            return Future.reject(
+                new SDKError(
+                    new Error("Current user is not authorized to rotate this group's private key as they are not a group administrator."),
+                    ErrorCodes.GROUP_PRIVATE_KEY_ROTATION_FAILURE
+                )
+            );
+        }
+        return UserApiEndpoints.callUserKeyListApi(group.adminIds)
+            .map((adminKeys) => adminKeys.result.map((user) => ({id: user.id, masterPublicKey: user.userMasterPublicKey})))
```

Comment:
We should probably add a check here to make sure that we got back the same number of public keys that we requested. If there's 10 admins of this group and we only get 7 public keys back from this call, that should probably cause a failure of the rotation because the group membership won't be the same before/after key rotation.

Reply:
@ernieturner The server will reject if the admins list doesn't match what's in the database. You can still check if you want, but shouldn't ever happen.

Reply:
Oh neat, good to know. Then ya, I'm fine with not having a check for that.